gomock.bzl

```diff
@@ -87,17 +62,200 @@ _gomock_sh = go_rule(
 def gomock(name, library, out, **kwargs):
     gopath_name = name + "_gomock_gopath"
     mockgen_tool = _MOCKGEN_TOOL
+
     if kwargs.get("mockgen_tool", None):
         mockgen_tool = kwargs["mockgen_tool"]
+
     go_path(
         name = gopath_name,
         deps = [library, mockgen_tool],
     )
 
-    _gomock_sh(
+    if kwargs.get("source", None):
+        _gomock_sh(
+            name = name,
+            library = library,
+            gopath_dep = gopath_name,
+            out = out,
+            **kwargs
+        )
+    else:
+        _gomock_reflect(
+            name = name,
+            library = library,
+            out = out,
+            mockgen_tool = mockgen_tool,
+            gopath_dep = gopath_name,
+            **kwargs
+        )
+
+def _gomock_reflect(name, library, out, mockgen_tool, gopath_dep, **kwargs):
+    interfaces = kwargs.get("interfaces", None)
+    mockgen_model_lib = _MOCKGEN_MODEL_LIB
+    if kwargs.get("mockgen_model_library", None):
+        mockgen_model_lib = kwargs["mockgen_model_library"]
+
+    prog_src = name + "_gomock_prog"
+    prog_src_out = prog_src + ".go"
+    _gomock_prog_gen(
+        name = prog_src,
+        interfaces = interfaces,
+        library = library,
+        package = kwargs.get("package", None),
+        out = prog_src_out,
+        mockgen_tool = mockgen_tool,
+        gopath_dep = gopath_dep,
+    )
+    prog_bin = name + "_gomock_prog_bin"
+    go_binary(
+        name = prog_bin,
+        srcs = [prog_src_out],
+        deps = [library, mockgen_model_lib],
+    )
+    _gomock_prog_exec(
```

Comment:
Is it really necessary to have the entire Go tools present in the build environment for this step? If I understand it correctly, the executable generated above is the only thing that's needed (assuming it is pure), right?

Reply:
Good point!

Reply:
Dang, I believe that roughly halved the elapse time.